src/DynamoCoreWpf/ViewModels/Core/DynamoViewModelEventArgs.cs

```diff
@@ -153,4 +159,142 @@ public OpenDocumentationLinkEventArgs(Uri link)
+        {
+            if (model == null) throw new ArgumentNullException(nameof(model));
+
+            Namespace = GetNamespace(model, dynamoViewModel);
+            Type = model.Name;
+            Description = model.Description;
+            Category = model.Category;
+            SetInputs(model);
+            SetOutputs(model);
+
+        }
+
+        private void SetOutputs(NodeModel nodeModel)
+        {
+            OutputNames = new List<string>();
+            OutputDescriptions = new List<string>();
+
+            var outputs = nodeModel.OutPorts;
+            foreach (var output in outputs)
+            {
+                OutputNames.Add(output.Name);
+                OutputDescriptions.Add(output.ToolTip);
+            }
+        }
+
+        private void SetInputs(NodeModel nodeModel)
+        {
+            InputNames = new List<string>();
+            InputDescriptions = new List<string>();
+
+            var inputs = nodeModel.InPorts;
+            foreach (var input in inputs)
+            {
+                InputNames.Add(input.Name);
+                InputDescriptions.Add(input.ToolTip);
+            }
+        }
+
+        static private string GetNamespace(NodeModel nodeModel, DynamoViewModel viewModel)
+        {
+            switch (nodeModel)
+            {
+                case Function function:
+                    var category = function.Category;
+                    var name = function.Name;
+                    if (CustomNodeHasCollisons(name, GetPackageName(nodeModel), viewModel))
+                    {
+                        var inputString = GetInputNames(function);
+                        return $"{category}.{name}({inputString})";
+                    }
+                    return $"{category}.{name}";
+
+                case DSFunctionBase dSFunction:
+                    var descriptor = dSFunction.Controller.Definition;
+                    var className = descriptor.ClassName;
+                    var functionName = descriptor.FunctionName;
+                    if (descriptor.IsOverloaded)
+                    {
+                        var inputString = GetInputNames(nodeModel);
+                        return $"{className}.{functionName}({inputString})";
+                    }
+
+                    return $"{className}.{functionName}";
+
+                case NodeModel node:
+                    var type = node.GetType();
+                    if (NodeModelHasCollisions(node.Name, viewModel))
+                    {
+                        return $"{type.FullName}({GetInputNames(nodeModel)})";
+                    }
+                    
+                    return type.FullName;
+                     
+
+                default:
+                    return string.Empty;
+            }
+        }
+
+        private static bool CustomNodeHasCollisons(string nodeName, string packageName, DynamoViewModel viewModel)
+        {
+            var pmExtension = viewModel.Model.GetPackageManagerExtension();
```

Comment:
I know this function gets used in lots of places carelessly, but it's possible it will return null, can we check its value before proceeding?